lib/Doctrine/KeyValueStore/Storage/AmazonDynamoDbStorage.php

```diff
@@ -0,0 +1,210 @@
+<?php
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license. For more information, see
+ * <http://www.doctrine-project.org>.
+ */
+
+namespace Doctrine\KeyValueStore\Storage;
+
+use Aws\DynamoDb\DynamoDbClient;
+use Aws\DynamoDb\Marshaler;
+use Doctrine\KeyValueStore\NotFoundException;
+
+class AmazonDynamoDbStorage implements Storage
+{
+    /**
+     * @var DynamoDbClient
+     */
+    private $client;
+
+    /**
+     * @var Marshaler
+     */
+    private $marshaler;
+
+    /**
+     * @var array
+     */
+    private $options = [
+        'default_key_name' => 'Id',
+        'storage_keys' => [],
+    ];
```

Comment:
Probably would be better to declare and define the properties `private $defaultKeyName = 'Id'` and `private $storageKeys = []`, for some reasons:
1. `$options` is used only by `AmazonDynamoDbStorage` so its content don't need to be flexible
2. if the final developer defines the *default_key_name* key as a non-scalar type, the class will generates different unexpected errors
3. a more strict type checking could be used asking for an `array $storageKeys = []`

What do you think about this @nicktacular?